<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Method for scalable identification of spatially variable genes 
    (SVGs) in spatially-resolved transcriptomics data. The method is based on 
    nearest-neighbor Gaussian processes and uses the BRISC algorithm for model 
    fitting and parameter estimation. Allows identification and ranking of SVGs 
    with flexible length scales across a tissue slide or within spatial domains 
    defined by covariates. Scales linearly with the number of spatial locations 
    and can be applied to datasets containing thousands or more spatial 
    locations.">
<title>Scalable identification of spatially variable genes in 
    spatially-resolved transcriptomics data • nnSVG</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="deps/Roboto-0.4.4/font.css" rel="stylesheet">
<link href="deps/JetBrains_Mono-0.4.4/font.css" rel="stylesheet">
<link href="deps/Roboto_Slab-0.4.4/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Scalable identification of spatially variable genes in 
    spatially-resolved transcriptomics data">
<meta property="og:description" content="Method for scalable identification of spatially variable genes 
    (SVGs) in spatially-resolved transcriptomics data. The method is based on 
    nearest-neighbor Gaussian processes and uses the BRISC algorithm for model 
    fitting and parameter estimation. Allows identification and ranking of SVGs 
    with flexible length scales across a tissue slide or within spatial domains 
    defined by covariates. Scales linearly with the number of spatial locations 
    and can be applied to datasets containing thousands or more spatial 
    locations.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">nnSVG</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.13</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/nnSVG.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lmweber/nnSVG/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="nnsvg">nnSVG<a class="anchor" aria-label="anchor" href="#nnsvg"></a>
</h1></div>

<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>nnSVG</code> is a method for scalable identification of spatially variable genes (SVGs) in spatially-resolved transcriptomics data.</p>
<p>The <code>nnSVG</code> method is based on nearest-neighbor Gaussian processes (<a href="https://www.tandfonline.com/doi/full/10.1080/01621459.2015.1044091" class="external-link">Datta et al., 2016</a>, <a href="https://www.tandfonline.com/doi/full/10.1080/10618600.2018.1537924" class="external-link">Finley et al., 2019</a>) and uses the BRISC algorithm (<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/sta4.184" class="external-link">Saha and Datta, 2018</a>) for model fitting and parameter estimation. <code>nnSVG</code> allows identification and ranking of SVGs with flexible length scales across a tissue slide or within spatial domains defined by covariates. The method scales linearly with the number of spatial locations and can be applied to datasets containing thousands or more spatial locations.</p>
<p><code>nnSVG</code> is implemented as an R package within the Bioconductor framework, and is available from <a href="https://bioconductor.org/packages/nnSVG" class="external-link">Bioconductor</a>.</p>
<p>Our preprint describing the method is available from <a href="https://www.biorxiv.org/content/10.1101/2022.05.16.492124v1" class="external-link">bioRxiv</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package can be installed from <a href="https://bioconductor.org/packages/nnSVG" class="external-link">Bioconductor</a> as follows, using R version 4.2 or above:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"nnSVG"</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, the latest development version of the package can also be installed from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"lmweber/nnSVG"</span><span class="op">)</span></span></code></pre></div>
<p>If you are installing from GitHub, the following dependency packages may also need to be installed manually from Bioconductor and CRAN:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SpatialExperiment"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"STexampleData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BRISC"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data-format">Input data format<a class="anchor" aria-label="anchor" href="#input-data-format"></a>
</h2>
<p>In the examples below, we assume the input data are provided as a <a href="https://bioconductor.org/packages/SpatialExperiment" class="external-link">SpatialExperiment</a> Bioconductor object. In this case, the outputs are stored in the <code>rowData</code> of the <code>SpatialExperiment</code> object.</p>
<p>However, the inputs can also be provided as a numeric matrix of normalized and transformed counts (e.g. log-transformed normalized counts, also known as logcounts) and a numeric matrix of spatial coordinates.</p>
<p>To provide the inputs as numeric matrices, please install the development version of the package from <a href="https://github.com/lmweber/nnSVG" class="external-link">GitHub</a> or the <code>devel</code> version of Bioconductor (which will become the new Bioconductor release version in October 2022).</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>An example workflow is provided in the package vignette from <a href="https://bioconductor.org/packages/nnSVG" class="external-link">Bioconductor</a> and shown here in short form.</p>
<p><strong>Load packages</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/nnSVG" class="external-link">nnSVG</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p><strong>Load example dataset</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example dataset from STexampleData package</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">Visium_humanDLPFC</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## [1] 33538  4992</span></span></code></pre></div>
<p><strong>Preprocessing</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># keep spots over tissue</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## [1] 33538  3639</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># spot-level quality control: already performed on this example dataset</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter low-expressed and mitochondrial genes</span></span>
<span><span class="co"># using function from nnSVG package with default filtering parameters</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/filter_genes.html">filter_genes</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Gene filtering: removing mitochondrial genes</span></span>
<span><span class="co">## removed 13 mitochondrial genes</span></span>
<span><span class="co">## Gene filtering: retaining genes with at least 3 counts in at least 0.5% (n = 19) of spatial locations</span></span>
<span><span class="co">## removed 30216 out of 33525 genes due to low expression</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate logcounts (log-transformed normalized counts) using scran package</span></span>
<span><span class="co"># using library size factors</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html" class="external-link">computeLibraryFactors</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="fu">assayNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## [1] "counts"    "logcounts"</span></span></code></pre></div>
<p><strong>Subset data for this example</strong></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select small set of random genes and several known SVGs for faster runtime in this example workflow</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">ix_random</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">known_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MOBP"</span>, <span class="st">"PCP4"</span>, <span class="st">"SNAP25"</span>, <span class="st">"HBB"</span>, <span class="st">"IGKC"</span>, <span class="st">"NPY"</span><span class="op">)</span></span>
<span><span class="va">ix_known</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">known_genes</span><span class="op">)</span></span>
<span><span class="va">ix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ix_known</span>, <span class="va">ix_random</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span><span class="va">ix</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## [1]   16 3639</span></span></code></pre></div>
<p><strong>Run nnSVG</strong></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set seed for reproducibility</span></span>
<span><span class="co"># run nnSVG using a single thread for this example workflow</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/nnSVG.html">nnSVG</a></span><span class="op">(</span><span class="va">spe</span>, n_threads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show results</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## DataFrame with 16 rows and 17 columns</span></span>
<span><span class="co">## [...]</span></span></code></pre></div>
<p><strong>Investigate results</strong></p>
<p>The results are stored in the <code>rowData</code> of the <code>SpatialExperiment</code> object.</p>
<p>The main results of interest are:</p>
<ul>
<li>
<code>LR_stat</code>: likelihood ratio (LR) statistics used to rank SVGs</li>
<li>
<code>rank</code>: rank of top SVGs according to LR statistics</li>
<li>
<code>pval</code>: approximate p-values</li>
<li>
<code>padj</code>: approximate p-values adjusted for multiple testing</li>
<li>
<code>prop_sv</code>: effect size defined as proportion of spatial variance</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># number of significant SVGs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">##     7     9</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show results for top n SVGs</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">rank</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">]</span>, <span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## DataFrame with 10 rows and 17 columns</span></span>
<span><span class="co">##                         gene_id   gene_name    feature_type   sigma.sq    tau.sq</span></span>
<span><span class="co">##                     &lt;character&gt; &lt;character&gt;     &lt;character&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## ENSG00000168314 ENSG00000168314        MOBP Gene Expression 1.38739383  0.364188</span></span>
<span><span class="co">## ENSG00000132639 ENSG00000132639      SNAP25 Gene Expression 0.43003959  0.430106</span></span>
<span><span class="co">## ENSG00000211592 ENSG00000211592        IGKC Gene Expression 0.56564845  0.455042</span></span>
<span><span class="co">## ENSG00000244734 ENSG00000244734         HBB Gene Expression 0.32942113  0.353754</span></span>
<span><span class="co">## ENSG00000183036 ENSG00000183036        PCP4 Gene Expression 0.23102220  0.452735</span></span>
<span><span class="co">## ENSG00000122585 ENSG00000122585         NPY Gene Expression 0.28567359  0.280173</span></span>
<span><span class="co">## ENSG00000129562 ENSG00000129562        DAD1 Gene Expression 0.02389607  0.464723</span></span>
<span><span class="co">## ENSG00000114923 ENSG00000114923      SLC4A3 Gene Expression 0.01147170  0.237260</span></span>
<span><span class="co">## ENSG00000133606 ENSG00000133606       MKRN1 Gene Expression 0.00632248  0.272432</span></span>
<span><span class="co">## ENSG00000143543 ENSG00000143543         JTB Gene Expression 0.07541566  0.463623</span></span>
<span><span class="co">##                        phi    loglik   runtime      mean       var     spcov</span></span>
<span><span class="co">##                  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## ENSG00000168314   1.102018  -3663.60     0.631  0.805525  1.205673  1.462248</span></span>
<span><span class="co">## ENSG00000132639   3.033847  -3912.70     0.450  3.451926  0.857922  0.189973</span></span>
<span><span class="co">## ENSG00000211592  20.107022  -4531.64     1.054  0.622937  1.007454  1.207340</span></span>
<span><span class="co">## ENSG00000244734  27.814098  -4044.96     1.559  0.411262  0.697673  1.395587</span></span>
<span><span class="co">## ENSG00000183036   8.272278  -4026.22     0.419  0.687961  0.684598  0.698656</span></span>
<span><span class="co">## ENSG00000122585  71.653290  -3995.23     0.843  0.393975  0.567383  1.356646</span></span>
<span><span class="co">## ENSG00000129562  10.141894  -3842.24     0.590  0.549318  0.489167  0.281410</span></span>
<span><span class="co">## ENSG00000114923  12.765645  -2617.36     0.658  0.250768  0.248816  0.427112</span></span>
<span><span class="co">## ENSG00000133606   0.082764  -2831.51     0.612  0.295404  0.278806  0.269171</span></span>
<span><span class="co">## ENSG00000143543 119.721419  -4036.28     0.731  0.654919  0.539172  0.419318</span></span>
<span><span class="co">##                   prop_sv loglik_lm    LR_stat      rank        pval        padj</span></span>
<span><span class="co">##                 &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">## ENSG00000168314 0.7920804  -5503.33 3679.46397         1 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000132639 0.4999614  -4884.19 1942.98556         2 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000211592 0.5541822  -5176.53 1289.77508         3 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000244734 0.4821910  -4507.99  926.04573         4 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000183036 0.3378716  -4473.57  894.68884         5 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000122585 0.5048609  -4131.87  273.27818         6 0.00000e+00 0.00000e+00</span></span>
<span><span class="co">## ENSG00000129562 0.0489053  -3861.98   39.49098         7 2.65854e-09 6.07667e-09</span></span>
<span><span class="co">## ENSG00000114923 0.0461207  -2632.02   29.31376         8 4.31119e-07 8.62238e-07</span></span>
<span><span class="co">## ENSG00000133606 0.0226812  -2839.08   15.15227         9 5.12539e-04 9.11181e-04</span></span>
<span><span class="co">## ENSG00000143543 0.1399077  -4039.07    5.59664        10 6.09124e-02 9.74599e-02</span></span></code></pre></div>
<p><strong>Plot expression of top SVG</strong></p>
<p>Plot expression of the top-ranked SVG.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot spatial expression of top-ranked SVG</span></span>
<span><span class="va">ix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">rank</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">ix_name</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="va">ix</span><span class="op">]</span></span>
<span><span class="va">ix_name</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## [1] "MOBP"</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, expr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="va">ix</span>, <span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pxl_col_in_fullres</span>, y <span class="op">=</span> <span class="va">pxl_row_in_fullres</span>, </span>
<span>               color <span class="op">=</span> <span class="va">expr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_reverse</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"gray90"</span>, high <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>                       trans <span class="op">=</span> <span class="st">"sqrt"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">expr</span><span class="op">)</span>, </span>
<span>                       name <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="va">ix_name</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span>, </span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://user-images.githubusercontent.com/8062417/179638201-a6d0cc21-a625-4899-8ab4-9b082b1d3a8c.png" alt="Spatial expression plot of top-ranked SVG" title="Spatial expression plot of top-ranked SVG" width="350px"></p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Our preprint describing <code>nnSVG</code> is available from bioRxiv:</p>
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2022.05.16.492124v1" class="external-link">Weber L.M. et al. (2022), “nnSVG: scalable identification of spatially variable genes using nearest-neighbor Gaussian processes”, bioRxiv</a></li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/nnSVG" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/lmweber/nnSVG/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/lmweber/nnSVG/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing nnSVG</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Lukas M. Weber <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3282-1730" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Stephanie C. Hicks <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-7858-0231" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lmweber/nnSVG/actions" class="external-link"><img src="https://github.com/lmweber/nnSVG/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Lukas M. Weber, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
